src/core/PrologInstance.ts

```diff
@@ -1,14 +1,16 @@
 import fs from "fs-extra";
+import path from "path";
 //@ts-ignore
 import swipl from 'swipl-stdio'
 //@ts-ignore
 import tmp from 'tmp'
+import { resourcePath } from "./Config";
 
 export default class PrologInstance {
     #engine
 
     constructor() {
-        this.#engine = new swipl.Engine()
+        this.#engine = new swipl.Engine(path.join(resourcePath, "swipl"))
```

Comment:
Why does this work on Linux as well without a change in this PR that copies (or creates a link of) the swipl binary to resourcePath?

Reply:
if you check the `swipl-stdio` package it doesn't use the swipl binary that we have, instead it would only use the binary if its in the $PATH